### ISSUE TITLE
Running unit test in coverage workflow is failing

### ISSUE THREAD

**Issue by josecelano** [torrust/torrust-tracker#904]
https://github.com/torrust/torrust-tracker/actions/runs/9562458284/job/26358873400?pr=897

Error: `Process completed with exit code 143.`

**Comment by josecelano** [comment]
Running it locally also makes my user's session (in ubuntu) crash:

```console
CARGO_INCREMENTAL="0" \
  RUSTFLAGS="-Z profile -C codegen-units=1 -C opt-level=0 -C link-dead-code -C overflow-checks=off -Z panic_abort_tests -C panic=abort" \
  RUSTDOCFLAGS="-Z profile -C codegen-units=1 -C opt-level=0 -C link-dead-code -C overflow-checks=off -Z panic_abort_tests -C panic=abort" \
  cargo test --tests --workspace --all-targets --all-features
```

It must be consuming a lot of resources.

**Comment by da2ce7** [comment]
closed by #931

**Comment by josecelano** [comment]
> closed by #931

Hey, @da2ce7, do you think this type of issue could also be fixed by extracting new packages in the workspace? I guess that would have two benefits:

- Faster building times
- Improved modularity
